src/mongocrypt-util.c

```diff
@@ -124,6 +124,34 @@ const char *mc_bson_type_to_string(bson_type_t t) {
     }
 }
 
+bool mc_is_valid_bson_type(int bson_type) {
```

Comment:
This function seems.... sus.

You seem to just be checking if the value is in the explicitly list of named numeric values.

Thing is that's not very meaningful in terms of if a type is "valid" or not.

`MINKEY`-`MAXKEY` are all "valid" as BSON types, for example.  Those two constants define the endpoints of the "User Defined BSONElements" range.

On the other side, EOD and UNDEFINED don't make sense as valid encodable types. (Though "Undefined" is defined, it's just no longer a valid type, so that one gets.... complicated).

TL;DR - The only truly "invalid" type is EOD.

Unless I'm misunderstanding the purpose of this method, I'd probably just reduce it to:
```c
return bson_type != BSON_TYPE_EOD;
```

Reply:
I went ahead and modified this in order to get a test passing, feel free to change it back if necessary.

Reply:
Huh, I didn't realize that `[MAXKEY, MINKEY]` are valid, I was wondering what those meant. Thanks for the clarification.